### PR TITLE
Fix codeLens for Run/Debug links showing up with latency

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

It fixes the issue https://github.com/Microsoft/vscode-java-debug/issues/438

### CODE REVIEW

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveMainMethodHandler.java [hunk 1]
```diff
@@ -43,6 +47,17 @@ public static Object resolveMainMethods(List<Object> arguments) throws DebugExce
             return Collections.emptyList();
         }
 
+        // When the current document is changed, the language server will receive a didChange request about the changed text and then
+        // trigger a background job to update the change to the CompilationUnit. Because of race condition, the resolveMainMethods may read
+        // an old CompilationUnit. So add some waiting logic to wait the Document Update to finish first.
+        try {
+            Job.getJobManager().join(DocumentLifeCycleHandler.DOCUMENT_LIFE_CYCLE_JOBS, new NullProgressMonitor());
```
Comment (andxu) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveMainMethodHandler.java]: Is there any chance to wait a smaller granularity of jobs of the current file?

Comment (testforstephen) [reply com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveMainMethodHandler.java]: @yaohaizh is this possible?

Comment (yaohaizh) [reply com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveMainMethodHandler.java]: Not sure.

Comment (testforstephen) [reply com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveMainMethodHandler.java]: Don't find the logic to wait on a smaller granularity job yet.

**Pull request merged**